### DISCUSSION TITLE
Update / Upgrade Path?

### DISCUSSION THREAD

**Discussion by Fefe-Hern** [bigcommerce/catalyst#708] [Q&A] [answered]
Interesting project to assist with a better headless interface for storefronts, but I'm wondering about how the Upgrade path will go, and if either codegen modules would be released, or instructions on what's changed between versions.

**Comment by chanceaclark** [comment, marked as answer]
Hey @Fefe-Hern!

We've been thinking about the upgrade process a bit and have some ideas in the works, but nothing concrete. We will be exploring codemods to get you at least 80% of the way there, but there might some manual intervention just like with any other software updates :( Regarding changelogs, we'll try and include migration paths within it, but we will try to also ship a migration guide with every "big" release to go along with a new version/tag of the core app.

Is there anything in particular that you had in mind that would be worth us exploring?

**Reply by Fefe-Hern** [reply]
Hey @chanceaclark thanks for the response.

Overall codegen with migration paths does sound good. I've been working on heavily (and I mean heavily) updated Cornerstone themes, and I've had to update via git-comparing and pulling in the desired changes after several major updates.

Cause of that after cloning the repo and seeing it's been updated since, I've been wondering if the best thing to do is to re-copy some files, or stay with the Catalyst I've cloned for a project.
